src/test/java/seedu/address/logic/commands/internships/task/EditInternshipTaskCommandTest.java

```diff
@@ -0,0 +1,160 @@
+import seedu.address.testutil.InternshipRoleBuilder;
+import seedu.address.testutil.InternshipTaskBuilder;
+
+public class EditInternshipTaskCommandTest {
+    private final InternshipRole role1 = new InternshipRoleBuilder().withName("Company1")
+            .withRole("SWE").withCycle("Summer").build();
+
+    private final InternshipRole role2 = new InternshipRoleBuilder().withName("Company2")
+            .withRole("SWE").withCycle("Winter").build();
+
+    private final InternshipTask task1BelongingToRole1 = new InternshipTaskBuilder()
+            .withInternshipRole(role1).withTaskName("Interview Preparation_1").build();
+
+    private final InternshipTask task2BelongingToRole1 = new InternshipTaskBuilder()
+            .withInternshipRole(role1).withTaskName("OA_1").build();
+
+    private final InternshipTask task1BelongingToRole2 = new InternshipTaskBuilder()
+            .withInternshipRole(role2).withTaskName("Interview Preparation_2").build();
+
+    private final InternshipTask task2BelongingToRole2 = new InternshipTaskBuilder()
+            .withInternshipRole(role2).withTaskName("OA_2").build();
+
+    private ModelManager model;
+
+    private AddressBook getAddressBook() {
+        AddressBook ab = new AddressBook();
+        ab.addInternshipRole(role1);
+        ab.addInternshipRole(role2);
+        ab.addInternshipTask(task1BelongingToRole1);
+        ab.addInternshipTask(task2BelongingToRole1);
+        ab.addInternshipTask(task1BelongingToRole2);
+        ab.addInternshipTask(task2BelongingToRole2);
+        return ab;
+    }
+
+    private void editTask(InternshipTask taskToBeEdited, InternshipTask editedTask, ModelManager model) {
+        model.setInternshipTask(taskToBeEdited, editedTask);
+    }
+
+    private EditInternshipTaskCommand getSampleEditCommand() {
+        return new EditInternshipTaskCommand(Index.fromOneBased(1),
+                new TaskOutcome(Outcome.AWAITING));
+    }
+
+    @BeforeEach
+    public void init() {
+        model = new ModelManager(getAddressBook(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_allFieldsProvided_taskEdited() {
+        Index index = Index.fromOneBased(1);
+        EditInternshipTaskCommand editCommand =
+                new EditInternshipTaskCommand(index, new TaskOutcome(Outcome.ACCEPTED));
+
+        ModelManager expectedModel = new ModelManager(getAddressBook(), new UserPrefs());
+
+        InternshipTask editedTask = new InternshipTaskBuilder().withInternshipRole(role1)
+                .withTaskName("Interview Preparation_1")
+                .withOutcome(Outcome.ACCEPTED).build();
+
+        editTask(task1BelongingToRole1, editedTask, expectedModel);
+
+        String expectedMessage = String.format(EditInternshipTaskCommand.MESSAGE_SUCCESS, Messages.format(editedTask));
+
+        assertCommandSuccess(editCommand, model, expectedMessage, expectedModel);
+    }
+
```

Comment:
Can add a test that checks when task outcome is edited to `offered` that `role` outcome is changed too